test/explorer.test.js

```diff
@@ -237,4 +264,14 @@ describe('explorer', function() {
     app.set('legacyExplorer', false);
     app.use(app.get('restApiRoot') || '/', loopback.rest());
   }
+
+  function createNewModel(app, explorerBase,modelName) {
+    var Model = loopback.PersistedModel.extend(modelName);
+    Model.attachTo(loopback.memory());
+    app.model(Model);
+
+    explorer(app, { mountPath: explorerBase });
+    app.set('legacyExplorer', false);
+    app.use(app.get('restApiRoot') || '/', loopback.rest());
```

Comment:
You should not need to call `explorer(app)` again, lines 274-275 should be removed as well.

When you call `explorer(app)` again, it will add a duplicate set of routes to the application. Each express route incurs a small performance penalty, therefore adding unnecessary duplicate routes can cause performance degradation.

Also the code adding new models should not need to be aware of the fact that the app is using explorer. Your current solution creates a coupling between unrelated parts of the system which is a bad practice.

So, please remove L273-L275, I am suggesting to move L269-L271 back to the test method and get rid of `createNewModel` entirely.